05-onchain.md

```diff
@@ -305,16 +305,19 @@ There are several possible cases for an offered HTLC:
    HTLC. In this case, the recipient must use the preimage, once it receives it
    from the outgoing HTLC; otherwise, it will lose funds by sending an outgoing
    payment without redeeming the incoming payment.
+4. The HTLC is forwarded as above, but the recipient also happens to be the
+   final destination for the payment.  In this case, it knows the preimage but
+   as it has not received the full payment, it must not reveal
+   it.<sup>[Preimage-Extraction](https://lists.linuxfoundation.org/pipermail/lightning-dev/2020-October/002857.html)</sup>.
+   This problem is most easily solved by only accepting preimages from
```

Comment:
I think "accepting" is unclear, if the preimages you're mentioning here are effectively the ones received for outgoing HTLCs, maybe it's better to say "relay preimages from". 

Note the current MUST NOT says "its own preimage" to dissociate from relayed preimages, but I concede it's confusing.